app/controllers/metrics_controller.rb

```diff
@@ -30,22 +31,30 @@ def create
   end
 
   def update
-    respond_to do |format|
-      if @metric.update(metric_params)
-        format.html { redirect_to @metric, notice: 'Metric was successfully updated.' }
-        format.json { render :show, status: :ok, location: @metric }
-      else
-        format.html { render :edit }
-        format.json { render json: @metric.errors, status: :unprocessable_entity }
+    if @metric.user == current_user
```

Comment:
Good to see the metrics being protected in `destroy` and `update`
```rb
if @metric.user == current_user
```
An alternative solution to protecting user-owned data is to use `current_user`-scoped queries. For example:

```
# Generated SQL query will contain a "AND metrics.user_id = ..." clause to 
# ensure only current_user-owned metrics will be returned:
@metric = current_user.metrics.find(params[:id])
```
At the moment the `edit` action doesn't have either of these protections which may be desirable.